crates/biome_js_analyze/src/analyzers/nursery/no_restricted_imports.rs

```diff
@@ -0,0 +1,77 @@
+use biome_analyze::context::RuleContext;
+use biome_analyze::{declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_js_syntax::inner_string_text;
+use biome_rowan::TextRange;
+use serde::{Deserialize, Serialize};
+
+use super::use_nodejs_import_protocol::AnyJsImportLike;
+
+declare_rule! {
+    /// Disallow specified modules when loaded by import or require.
+    ///
+    /// https://eslint.org/docs/latest/rules/no-restricted-imports
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```json
+    /// {
+    ///     "noRestrictedImports": {
+    ///         "options": {
+    ///             "paths": ["lodash"]
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// const lodash = require("lodash");
+    /// ```
+    pub(crate) NoRestrictedImports {
+        version: "next",
+        name: "noRestrictedImports",
+        source: RuleSource::Eslint("no-restricted-imports"),
+        recommended: false,
+    }
+}
+
+/// Options for the rule `noRestrictedImports`.
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct RestrictedImportsOptions {
+    /// A list of names that should trigger the rule
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    paths: Vec<String>,
+}
+
+impl Rule for NoRestrictedImports {
+    type Query = Ast<AnyJsImportLike>;
+    type State = (TextRange, String);
+    type Signals = Option<Self::State>;
+    type Options = RestrictedImportsOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let module_name = ctx.query().module_name_token()?;
+        let options = ctx.options();
+        let trimmed_name = inner_string_text(&module_name).to_string();
+
+        options
+            .paths
+            .contains(&trimmed_name)
+            .then_some((module_name.text_trimmed_range(), trimmed_name))
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, (span, text): &Self::State) -> Option<RuleDiagnostic> {
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            *span,
+            markup! {
+                "Importing "<Emphasis>{text}</Emphasis>" is forbidden."
```

Comment:
I wonder if we should use `disabled` instead of `forbidden`. Any opinion?

Reply:
Disabled is a better choice, I agree.